### PR TITLE
[fix](fe)fix bug if left table is empty and there are multiple right tables need do bucket shuffle to left side

### PR DESCRIPTION
**Pull request by starocean999**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.04 seconds
 stream load tsv:          427 seconds loaded 74807831229 Bytes, about 167 MB/s
 stream load json:         23 seconds loaded 2358488459 Bytes, about 97 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230427114015_clickbench_pr_136529.html

**Review by morrySnow — DISMISSED** [review]

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java [hunk 1]
```diff
@@ -1357,8 +1357,8 @@ private void computeFragmentExecParams() throws Exception {
 
                 // when left table is empty, it's bucketset is empty.
                 // set right table destination address to the address of left table
-                if (destParams.instanceExecParams.size() == 1
-                        && destParams.instanceExecParams.get(0).bucketSeqSet.isEmpty()) {
+                if (destParams.instanceExecParams.size() == 1 && (bucketNum == 0
+                        || destParams.instanceExecParams.get(0).bucketSeqSet.isEmpty())) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:1361]: why add back `destParams.instanceExecParams.get(0).bucketSeqSet.isEmpty()`?

Comment (starocean999) [reply fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:1361]: for colocate join, the bucketNum is the bucket number in default distribution info. So even the table is empty, the bucketNum is not 0. So add it back to handle the colocate join case

**Review by englefly — APPROVED** [review]

**Pull request merged**